modules/ROOT/pages/build-application-from-api.adoc

```diff
@@ -0,0 +1,198 @@
+
+[NOTE]
+MuleSoft recommends Design Center for creating and syncing RAML code; however, Anypoint Studio includes a RAML editor that provides syntax validation and suggestions if you want to perform the entire development within Studio.
+
+== APIkit Anatomy
+A REST or SOAP project that uses APIkit has the following parts:
+
+* A RAML or WSDL interface.
+* A main flow
+* Backend flows
+
+.REST Studio Project.
+image::mruntime-rest-studio-project.png[REST Studio Project]
+
+.SOAP Studio Project.
+image::mruntime-soap-studio-project.png[SOAP Studio Project]
+
+=== API Interface
+The interface defines the API, designating resources that contain or access data and specify actions, such as GET and PUT, on the example data
+
+==== REST Interface
+The relationship between REST resources and actions is called resource-action pairing. The interface exposes internal data assets requested by the REST service.
+
+==== SOAP Interface
+The interface defined by a WSDL file exposes a number of services. Each service has a set of operations. The underlying type of data is XML defined by schemas within the WSDL file or by an imported XSD file.
+
+=== Main Flow
+The main flow for these APIs manages these functions:
+
+* Exposes the API using HTTP or Jetty.
+* Routes requests between the interface and the backend flows based on the HTTP request.
+* Accepts requests, sends responses, and facilitates communication between the interface and API processes.
+
+For REST APIs the api-main flow consists of HTTP Listener and APIkit Router components. The APIkit router is a key message processor that validates requests against the RAML definition, enriches messages, for example by adding default values to the messages, and routes requests to a flow. "Bad request" is returned if the request is invalid, for example, and "Not implemented" is returned if the RAML resource that you request is not associated with a flow.
+
+.REST API Main Flow.
+image::mruntime-rest-main-flow.png[REST API Main Flow]
+
+For SOAP APIs the main-api flow consists of HTTP Listener and SOAP Router components.  Based on a WSDL file, the SOAP Router component routes arriving calls to the corresponding flow depending on the resource and method.
+
+.REST API Main Flow.
+image::mruntime-soap-main-flow.png[SOAP API Main Flow]
+
+=== Backend Flows
+APIkit for REST generates a backend flow for each resource-action pairing in a RAML file. +
+APIkit for SOAP generates a backend flow for each operation in a WSDL. +
+In the following example, the RAML interface receives the GET request users. The backend flow accesses a database to look up users by id.
+
+.REST API Backend Flows.
+image::mruntime-rest-backend-flows.png[REST API Backend Flows]
+
+=== Simulating Calls
+For REST APIs, use the API Console to simulate calls to the API. +
+For SOAP APIs, use a SOAP testing tool such as SoapUI, to simulate calls to the API.
```

Comment:
to simulate calls to the API. -> they are real calls

Reply:
Updated.